src/sandstorm/web-session.capnp

```diff
@@ -366,6 +375,27 @@ interface WebSession @0xa50711a14d35a8ce extends(Grain.UiSession) {
       # TODO(someday):  Return blob directly from storage, so data doesn't have to stream through
       #   the app?
     }
+
+    additionalHeaders @19 :List(Header);
+    # Additional headers present in the reponse. Only whitelisted headers are
+    # permitted.
+
+    struct Header {
+      name @0 :Text;  # lower-cased name
+      value @1 :Text;
+      prefixed @2 :Bool = false;
+      # Indicates whether name represents a header that was prepended with appHeaderPrefix.
+      # It should be reconstructed with the prefix before being sent to the HTTP client.
+    }
+
+    const headerWhitelist :List(Text) = [
+      # Non-standard response headers which are whitelisted for backwards-compatibility
+      # purposes. This whitelist exists to help avoid the need to modify code originally written
+      # without Sandstorm in mind -- especially to avoid modifying client apps.
+      # Feel free to send us pull requests adding additional headers.
+      "allow"
```

Comment:
I'd prefer only to whitelist non-standard headers. For standard headers, we should represent the header's meaning semantically using regular struct fields.

If the purpose is just for testing, perhaps you could make up a header `X-Sandstorm-Whitelist-Test`, or maybe add `X-OC-MTime` as requested by @mnutt.